File: Java/opentraceability/src/main/java/opentraceability/gdst/events/GDSTComminglingEvent.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.models.events.TransformationEventBase;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;

public class GDSTComminglingEvent extends TransformationEventBase<GDSTILMD> implements IGDSTEvent {
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "productOwner")
    @OpenTraceabilityJsonAttribute(name = "gdst:productOwner")
    private PGLN productOwner;
    @Override
    public PGLN getProductOwner() {
        return this.productOwner;
    }

    // Add constructor, getters, and setters here
    public GDSTComminglingEvent()
    {
        this.businessStep = URI.create("urn:gdst:bizStep:commingling");
    }
}

File: CSharp/OpenTraceability.GDST/Events/GDSTComminglingEvent.cs
using OpenTraceability.Models.Events;
using OpenTraceability.Models.Identifiers;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.GDST.Events
{
    public class GDSTComminglingEvent: TransformationEvent<GDSTILMD>, IGDSTILMDEvent   
    {
        [OpenTraceability(Constants.GDST_NAMESPACE, "productOwner")]
        [OpenTraceabilityJson("gdst:productOwner")]
        public PGLN? ProductOwner { get; set; }

        public GDSTComminglingEvent()
        {
            this.BusinessStep = new Uri("urn:gdst:bizStep:commingling");
        }
    }
}
